mailbox/requirements.in

```diff
@@ -0,0 +1 @@
+magic-wormhole-mailbox-server -e git+https://github.com/magic-wormhole/magic-wormhole-mailbox-server/tarball/4b358859ba80de37c3dc0a5f67ec36909fd48234#egg=magic-wormhole-mailbox-server
```

Comment:
I think these github hashes (of magically-produced tarballs) are not stable...?

Reply:
That hash is the commit hash from the repo:

https://github.com/magic-wormhole/magic-wormhole-mailbox-server/commit/4b358859ba80de37c3dc0a5f67ec36909fd48234

Pretty stable, you can also use a tag (but this is less stable).

Please note that the source URL did not change with this PR, we already used it before.

Reply:
Maybe it's a different thing, but there was some issue with using GitHub links (e.g. from the "Download" button, approximately). Maybe it was only for ZIPs?

Reply:
Right, okay that was it -- that the bits can change (so if you have a workflow that checks those, it'll fail -- but only "somtimes" e.g. when GitHub nukes a cache or something?)

Reply:
Can't we use the git hash to access the archive (w/o relying on git)?
Are you suggesting there could be some differences between sources cloned using Git CLI and those "mirrored" with plain HTTP(s)?

I was under the impression that `tar` and `zip` archives were part of Git itself (e.g.: `git archive`)!

Reply:
> I was under the impression that tar and zip archives were part of Git itself (e.g.: git archive)!

I believe that's correct, but the output isn't stable. So if you have a process that checks hashes, it'll fail sometimes. (Certainly there are ways besides "use the github API" to get the sources of course, I was just highlighting that that particular method has a potential problem).

I believe this discusses the problem: https://github.blog/2023-02-21-update-on-the-future-stability-of-source-code-archives-and-hashes/

Reply:
I don't believe that `pip freeze` incantation is doing anything to check the contents...? (e.g. like `--require-hahes` in `pip` might)